src/content/articles/for-in-string-types.mdx

```diff
@@ -0,0 +1,212 @@
+---
+date: "February 25 2023"
+description: "Why TypeScript reports a `string` type instead of `keyof typeof` for some forms of loops over objects."
+meta: keyof, loops, objects, strings
+---
+
+# Why `for...in` and `Object.keys()` Give `string` Types
+
+Why does TypeScript report a `string` type instead of `keyof typeof` the object when you use a `for...in` loop or `Object.keys()` on an object?
+
+One of the most commonly asked questions in TypeScript help forums is why code like the following gives an error under the [`noImplicitAny` compiler option](https://www.typescriptlang.org/tsconfig#noImplicitAny).
```

Comment:
[Nitpick/proofreading] Hmmm..., there seems to be some ambiguity in this part of your article. Maybe you can try the suggestion below to make it clearer.  
```suggestion
 One of the most commonly asked questions in TypeScript help forums is why code like the following gives an error under the [`noImplicitAny` compiler option](https://www.typescriptlang.org/tsconfig#noImplicitAny).
```